hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -879,6 +1244,169 @@ def __repr__(self):
             ", frozen" if self.frozen else "",
         )
 
+    def draw_integer(
+        self,
+        *,
+        min_value: Optional[int] = None,
+        max_value: Optional[int] = None,
+        # weights are for choosing an element index from a bounded range
+        weights: Optional[Sequence[float]] = None,
+        shrink_towards: int = 0,
+    ):
+        # Validate arguments
+        if weights is not None:
+            assert min_value is not None
+            assert max_value is not None
+            assert (max_value - min_value) <= 1024  # arbitrary practical limit
+
+        return self.provider.draw_integer(
+            min_value=min_value,
+            max_value=max_value,
+            weights=weights,
+            shrink_towards=shrink_towards,
+        )
+
+    def draw_float(
+        self,
+        *,
+        min_value: Optional[float] = None,
+        max_value: Optional[float] = None,
+        allow_nan: bool = True,
+        smallest_nonzero_magnitude: Optional[float] = None,
+        # TODO: consider supporting these float widths at the IR level in the
+        # future.
+        # width: Literal[16, 32, 64] = 64,
+        # exclude_min and exclude_max handled higher up
+    ):
+        # FIXME assertions about forced w.r.t min_value, max_value, allow_nan
+        return self.provider.draw_float(
+            min_value=min_value,
+            max_value=max_value,
+            allow_nan=allow_nan,
+            smallest_nonzero_magnitude=smallest_nonzero_magnitude,
+        )
+
+    def draw_string(
+        self,
+        intervals: IntervalSet,
+        *,
+        min_size: int = 0,
+        max_size: Optional[int] = None,
+    ) -> str:
+        return self.provider.draw_string(
+            intervals, min_size=min_size, max_size=max_size
+        )
+
+    def draw_bytes(self, size: int):
+        return self.provider.draw_bytes(size)
+
+    def draw_boolean(self, p: float = 0.5, *, forced: Optional[bool] = None):
+        return self.provider.draw_boolean(p, forced=forced)
+
+    def draw_character(self, intervals: IntervalSet) -> chr:
```

Comment:
I think all you were pointing out by the latter comment is that `_draw_character` can now just return `intervals.char_in_shrink_order(i)`? I don't see much duplication otherwise. But correct me if I'm wrong.